velox/dwio/common/SelectiveColumnReader.h

```diff
@@ -275,7 +275,17 @@ class SelectiveColumnReader {
   void initReturnReaderNulls(RowSet rows);
 
   void setNumValues(vector_size_t size) {
-    numValues_ = size;
+    numValues_ = numScanned_ + size;
+  }
+
+  void setNumScanned(vector_size_t numScanned) {
```

Comment:
I don't think you need to store it in column reader.  It can be a separate object from both column reader and column visitor.  For formats that does not need paging, this is useless.

Reply:
FYI in Nimble I am passing it around as extra parameter in `readWithVisitor` in decoders

Reply:
I evaluated it and felt that if `numScanned` was moved to the Parquet-related implementation class, it would be difficult to implement other methods because `numScanned` is also used in many `facebook::velox::dwio::common` classes. For example, `facebook::velox::dwio::common::DirectDecoder`.

Reply:
Let's try pass it as parameter to `readWithVisitor`.   So far I see it is only used in `DirectDecoder::readWithVisitor`.

Reply:
This code has been completely refactored, see https://github.com/facebookincubator/velox/pull/10468/commits/1a633bd889898f1cd82801b7727eb161bf5acb4b